### PR TITLE
HS-54: sign up screen

### PR DESCRIPTION
**Pull request by dlvinh**
Create Sign up screen with shared button theme.
Widget tests are implemented in both SignUpScreen widget and shared buttons widget

### CODE REVIEW

**Review by dlvinh — COMMENTED** [review]
Screenshot 2022-11-20 at 00 21 54

File: lib/features/sign_up/view/sign_up_view.dart [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                          padding: const EdgeInsets.only(bottom: 16),
+                          child: SecondaryButton(
+                            label: "Sign up with Google",
+                            iconURL: Assets.images.google,
+                            onPressed: () {},
+                          )),
+                      Padding(
+                          padding: const EdgeInsets.only(bottom: 16),
+                          child: SecondaryButton(
+                            label: "Sign up with Facebook",
```
Comment (hellomentorplus) [inline lib/features/sign_up/view/sign_up_view.dart:37]: use localization text

File: lib/initial_app.dart [hunk 2]
```diff
@@ -20,7 +21,7 @@ class MyApp extends StatelessWidget {
           )
           // TODO add your bloc creation here
         ],
-        child: const HomePageView(),
+        child: const SignUpScreen(),
```
Comment (hellomentorplus) [inline lib/initial_app.dart:24]: first page is home page

**Review by hellomentorplus — CHANGES_REQUESTED** [review]

File: lib/features/sign_up/view/sign_up_view.dart (hunk 1 above)
Comment (dlvinh) [reply lib/features/sign_up/view/sign_up_view.dart]: Already added in the latest commit

File: lib/initial_app.dart (hunk 2 above)
Comment (dlvinh) [reply lib/initial_app.dart]: Already changed

**Comment by dlvinh** [comment]
Screenshot 2022-12-01 at 21 47 09

File: lib/features/sign_up/view/sign_up_view.dart [hunk 3]
```diff
@@ -13,83 +16,101 @@ class SignUpScreen extends StatelessWidget {
...
-                          // NAVIGATE TO HOME
-                        },
-                      ),
-                      const SizedBox(
-                        height: 71,
-                      )
-                    ]))
-              ],
-            )));
+    return MaterialApp(
```
Comment (hellomentorplus) [inline lib/features/sign_up/view/sign_up_view.dart:19]: why do u start another MaterialApp here?

**Review by hellomentorplus — CHANGES_REQUESTED** [review]
why do u start a new MaterialApp in SignUp screen?

File: lib/features/sign_up/view/sign_up_view.dart [hunk 4]
```diff
@@ -1 +1,102 @@
...
+                  Navigator.push(context,
+                      MaterialPageRoute(builder: (context) => HomePageView()));
+                },
+              ),
+            )),
+        body: Padding(
+            padding: const EdgeInsets.only(top: 60, left: 16, right: 16),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              crossAxisAlignment: CrossAxisAlignment.stretch,
```
Comment (hellomentorplus) [inline lib/features/sign_up/view/sign_up_view.dart:38]: remove this column

File: lib/features/sign_up/view/sign_up_view.dart [hunk 5]
```diff
@@ -1 +1,102 @@
...
+                                        fontSize: 12, color: HSColor.onSurface),
+                                    text: "Already have account ? "),
+                                TextSpan(
+                                    style: const TextStyle(
+                                        fontWeight: FontWeight.w600,
+                                        decoration: TextDecoration.underline),
+                                    text: "Sign in",
+                                    recognizer: TapGestureRecognizer()
+                                      ..onTapDown = (details) {
+                                        print("Tap down event");
```
Comment (hellomentorplus) [inline lib/features/sign_up/view/sign_up_view.dart:83]: avoid print in production

File: lib/features/sign_up/view/sign_up_view.dart [hunk 6]
```diff
@@ -1 +1,102 @@
...
+                                      })
+                              ]))),
+                      TextOnlyButton(
+                        label: "Skip",
```
Comment (hellomentorplus) [inline lib/features/sign_up/view/sign_up_view.dart:87]: move into localization

File: lib/features/sign_up/view/sign_up_view.dart [hunk 7]
```diff
@@ -1 +1,102 @@
...
+                                  children: [
+                                TextSpan(
+                                    style: textTheme.bodyMedium?.copyWith(
+                                        fontSize: 12, color: HSColor.onSurface),
+                                    text: "Already have account ? "),
+                                TextSpan(
+                                    style: const TextStyle(
+                                        fontWeight: FontWeight.w600,
+                                        decoration: TextDecoration.underline),
+                                    text: "Sign in",
```
Comment (hellomentorplus) [inline lib/features/sign_up/view/sign_up_view.dart:80]: move into localization

File: lib/features/sign_up/view/sign_up_view.dart [hunk 8]
```diff
@@ -1 +1,102 @@
...
+                          padding: const EdgeInsets.only(bottom: 16),
+                          child: SecondaryButton(
+                            label: HatSpaceStrings.of(context)!.facebookSignUp,
+                            iconURL: Assets.images.facebookround,
+                            onPressed: () {},
+                          )),
+                      Padding(
+                          padding: const EdgeInsets.only(bottom: 18),
+                          child: SecondaryButton(
+                            label: HatSpaceStrings.of(context)!.emailSignUp,
```
Comment (hellomentorplus) [inline lib/features/sign_up/view/sign_up_view.dart:61]: remove `!`

File: pubspec.yaml [hunk 9]
```diff
@@ -29,7 +29,8 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-
+  flutter_localizations: # Add this line
```
Comment (hellomentorplus) [inline pubspec.yaml:32]: refer to our confluence how we use localization

**Review by hellomentorplus — CHANGES_REQUESTED** [review]

File: lib/features/sign_up/view/sign_up_view.dart (hunk 3 above)
Comment (dlvinh) [reply lib/features/sign_up/view/sign_up_view.dart]: Already fixed

File: lib/features/sign_up/view/sign_up_view.dart [hunk 10]
```diff
@@ -1 +1,102 @@
...
+        appBar: AppBar(
+            elevation: 0, // Remove shadow from app bar background
+            backgroundColor: HSColor.background,
+            leading: Padding(
+              padding: const EdgeInsets.only(top: 0, left: 10, right: 0),
+              child: IconButton(
+                icon: SvgPicture.asset(Assets.images.closeIcon),
+                onPressed: () {
+                  // ON PRESSED EVENT HERE
+                  Navigator.push(context,
```
[the remark this answers is not in the document]
Comment (dlvinh) [reply lib/features/sign_up/view/sign_up_view.dart]: already changed

File: pubspec.yaml (hunk 9 above)
Comment (dlvinh) [reply pubspec.yaml]: already remove

**Review by suesitran — DISMISSED** [review]
good job!

**Review by hellomentorplus — DISMISSED** [review]

File: lib/main.dart [hunk 11]
```diff
@@ -5,9 +5,9 @@ import 'package:hatspace/firebase_options.dart';
 import 'package:hatspace/initial_app.dart';
 import 'package:hatspace/singleton/hs_singleton.dart';
 
+bool? showFirstSignUp;
```
Comment (hellomentorplus) [inline lib/main.dart:8]: please remove this and include in next PR

**Review by suesitran — APPROVED** [review]

**Pull request merged**